exchanges/okx/okx_websocket.go

```diff
@@ -0,0 +1,2055 @@
+	case okxChannelGridPositions:
+		var response WsContractGridAlgoOrder
+		return ok.wsProcessPushData(respRaw, &response)
+	case okcChannelGridSubOrders:
+		var response WsGridSubOrderData
+		return ok.wsProcessPushData(respRaw, &response)
+	case okxChannelInstruments:
+		var response WSInstrumentResponse
+		return ok.wsProcessPushData(respRaw, &response)
+	case okxChannelOpenInterest:
+		var response WSOpenInterestResponse
+		return ok.wsProcessPushData(respRaw, &response)
+	case okxChannelTrades:
+		return ok.wsProcessTrades(respRaw)
+	case okxChannelEstimatedPrice:
+		var response WsDeliveryEstimatedPrice
+		return ok.wsProcessPushData(respRaw, &response)
+	case okxChannelMarkPrice,
+		okxChannelPriceLimit:
+		var response WsMarkPrice
+		return ok.wsProcessPushData(respRaw, &response)
+	case okxChannelOrderBooks,
+		okxChannelOrderBooks5,
+		okxChannelOrderBooks50TBT,
+		okxChannelBBOTBT,
+		okxChannelOrderBooksTBT:
+		return ok.wsProcessOrderBooks(respRaw)
+	case okxChannelOptSummary:
+		var response WsOptionSummary
+		return ok.wsProcessPushData(respRaw, &response)
+	case okxChannelFundingRate:
+		var response WsFundingRate
+		return ok.wsProcessPushData(respRaw, &response)
+	case okxChannelMarkPriceCandle1Y, okxChannelMarkPriceCandle6M, okxChannelMarkPriceCandle3M, okxChannelMarkPriceCandle1M,
+		okxChannelMarkPriceCandle1W, okxChannelMarkPriceCandle1D, okxChannelMarkPriceCandle2D, okxChannelMarkPriceCandle3D,
+		okxChannelMarkPriceCandle5D, okxChannelMarkPriceCandle12H, okxChannelMarkPriceCandle6H, okxChannelMarkPriceCandle4H,
+		okxChannelMarkPriceCandle2H, okxChannelMarkPriceCandle1H, okxChannelMarkPriceCandle30m, okxChannelMarkPriceCandle15m,
+		okxChannelMarkPriceCandle5m, okxChannelMarkPriceCandle3m, okxChannelMarkPriceCandle1m, okxChannelMarkPriceCandle1Yutc,
+		okxChannelMarkPriceCandle3Mutc, okxChannelMarkPriceCandle1Mutc, okxChannelMarkPriceCandle1Wutc, okxChannelMarkPriceCandle1Dutc,
+		okxChannelMarkPriceCandle2Dutc, okxChannelMarkPriceCandle3Dutc, okxChannelMarkPriceCandle5Dutc, okxChannelMarkPriceCandle12Hutc,
+		okxChannelMarkPriceCandle6Hutc:
+		return ok.wsHandleMarkPriceCandles(respRaw)
+	default:
+		ok.Websocket.DataHandler <- stream.UnhandledMessageWarning{Message: ok.Name + stream.UnhandledMessage + string(respRaw)}
+		return nil
+	}
+}
+
+// wsProcessIndexCandles processes index candlestic data
+func (ok *Okx) wsProcessIndexCandles(intermediate *wsIncomingData) error {
+	if intermediate == nil {
+		return errNilArgument
+	}
+	var response WSCandlestickResponse
+	if len(intermediate.Data) == 0 {
+		return errNoCandlestickDataFound
+	}
+	pair, err := ok.GetPairFromInstrumentID(intermediate.Argument.InstrumentID)
+	if err != nil {
+		return err
+	}
+	var a asset.Item
+	a, _ = ok.GetAssetTypeFromInstrumentType(intermediate.Argument.InstrumentType)
+	candleInterval := strings.TrimPrefix(intermediate.Argument.Channel, candle)
+	for i := range response.Data {
+		candles, okay := (intermediate.Data[i]).([5]string)
```

Comment:
Rename var, conflicts with `candles` var. Even though this still works, it makes the code less readable